### PR TITLE
Feature/bmauer/fixes #1789

### LINKED ISSUE #1789: New feature in ExtData needed for GAAS, allow missing files
Just a placeholder issue so I can reference the issue # in the PR. Apparently GAAS has moved to use ExtData and this was not tested in the GEOSadas.  So to get it to working 1 new feature is needed:

Since GAAS only needs the files at certain times, just like the IAU files, like if the GEOSadas starts at 21z, it actually need the 00z file to do the analysis and reads this at 00z, then needs no new files during the forecast part of the segment. This is very different than when this is run in replay mode. So basically we need to allow for ExtData allow the absence of a file in the sequence to not cause a failure.

### PR DESCRIPTION
**Pull request by bena-nasa**
This is a hurry job to address #1789
If you have a non-climatology, allow gaps in the datasets, just set the whole bracket field to undef, then the pointer we are updating becomes undef entirely itself.

I added a function to test if a field is identically MAPL undef they can use  in GAAS to make the determination if data is available to perform the whatever it is GAAS does or not.

Added a new test, and did a bit of cleanup with the ExtData tests.

Starting to get ugly, needs some cleanup when there is time.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [X] I have tested this change with a run of GEOSgcm (if non-trivial)
- [X] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [X] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: base/FieldUtilities.F90 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+implicit none
+private
+
+public check_if_field_undef
+
+contains
+
+function check_if_field_undef(field,rc) result(field_is_undef)
+   logical :: field_is_undef
```
Comment (tclune) [inline base/FieldUtilities.F90:16]: This name is not well suited to a function as opposed to a subroutine.   I prefer something like:
```suggestion
function is_field_undef(field,rc) result(field_is_undef)
   logical :: is_field_undef
```

**Review by tclune — CHANGES_REQUESTED** [review]
I prefer a different name for the new function.  Not wed to my suggestion but should be a name that suggests something is returned.

Quibble, quibble, quibble

**Comment by mathomp4** [comment]
Looks like there is a name collision:

```
FAILED: gridcomps/ExtData2G/CMakeFiles/MAPL.ExtData2G.dir/ExtDataSimpleFileHandler.F90.o include/MAPL.ExtData2G/mapl_extdatasimplefilehandler.mod 
/gcc/bin/g******* -I/root/project/MAPL/gridcomps/ExtData2G -I/root/project/MAPL/gridcomps/ExtData2G -I/root/project/workspace/build-MAPL/gridcomps/ExtData2G -I/root/project/workspace/build-MAPL/include/MAPL.ExtData2G -I/root/project/MAPL/include -I/root/project/MAPL/shared -I/root/project/workspace/build-MAPL/shared -I/root/project/workspace/build-MAPL/include/MAPL.shared -I/root/project/MAPL/shared/Constants -I/root/project/workspace/build-MAPL/shared/Constants -I/root/project/workspace/build-MAPL/include/MAPL.constants -I/root/project/MAPL/base -I/root/project/workspace/build-MAPL/base -I/root/project/workspace/build-MAPL/include/MAPL.base -I/root/project/MAPL/profiler -I/root/project/workspace/build-MAPL/profiler -I/root/project/workspace/build-MAPL/include/MAPL.profiler -I/root/project/MAPL/pfio -I/root/project/workspace/build-MAPL/pfio -I/root/project/workspace/build-MAPL/include/MAPL.pfio -I/root/project/MAPL/MAPL_cfio -I/root/project/workspace/build-MAPL/MAPL_cfio_r4 -I/root/project/workspace/build-MAPL/include/MAPL_cfio_r4 -I/root/project/MAPL/generic -I/root/project/workspace/build-MAPL/generic -I/root/project/workspace/build-MAPL/include/MAPL.generic -I/root/project/MAPL/oomph -I/root/project/workspace/build-MAPL/oomph -I/root/project/workspace/build-MAPL/include/MAPL.oomph -I/root/project/MAPL/griddedio -I/root/project/workspace/build-MAPL/griddedio -I/root/project/workspace/build-MAPL/include/MAPL.griddedio -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/GFTL_SHARED-1.5/include/v1 -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/GFTL-1.8/include/v1 -I/opt/openmpi-4.1.4/include -I/opt/openmpi-4.1.4/lib -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/PFLOGGER-1.9/include -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/YAFYAML-1.0/include -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/GFTL-1.8/include/v2 -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/GFTL_SHARED-1.5/include/v2 -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/include/netcdf -I/baselibs/v7.5.0/gcc_12.1.0-openmpi_4.1.4/Linux/include/esmf -O0 -g -fcheck=all,no-array-temps -finit-real=snan -save-temps -ffree-line-length-none -fno-range-check -Wno-missing-include-dirs -fbacktrace -Wno-unused-dummy-argument -ffpe-trap=zero,overflow -fbacktrace -fallow-argument-mismatch -fallow-invalid-boz -falign-commons -Jinclude/MAPL.ExtData2G -fPIC -pthread -fpreprocessed -c gridcomps/ExtData2G/CMakeFiles/MAPL.ExtData2G.dir/ExtDataSimpleFileHandler.F90-pp.f90 -o gridcomps/ExtData2G/CMakeFiles/MAPL.ExtData2G.dir/ExtDataSimpleFileHandler.F90.o
/root/project/MAPL/gridcomps/ExtData2G/ExtDataSimpleFileHandler.F90:20:49:

   15 |    use MAPL_ExtDataConstants
      |       2                                          
......
   20 |    character(len=14), parameter :: file_not_found = "file_not_found"
      |                                                 1
Error: Symbol 'file_not_found' at (1) conflicts with symbol from module 'mapl_extdataconstants', use-associated at (2)
```

**Comment by bena-nasa** [comment]
I put a DNA on this PR. Realized when trying this with GAAS my test was incomplete in that I started the iteration of the driver that exercised ExtData while there was still data, then it ran through the time period that had the gap in the data.
But, if you start during that gap (which is what it would look like for GAAS in the DAS) I need to do additional work to find A file as I need the metadata from some file in sequence during initialization

**Review by tclune — APPROVED** [review]

**Pull request merged**